collect_app/src/main/java/org/odk/collect/android/formlists/savedformlist/SavedFormListViewModel.kt

```diff
@@ -65,14 +68,20 @@ class SavedFormListViewModel(
 
     fun deleteForms(databaseIds: LongArray): LiveData<Consumable<Int>?> {
         val result = MutableLiveData<Consumable<Int>?>(null)
-        worker.immediate(
-            background = {
-                databaseIds.forEach { instancesDataService.deleteInstance(it) }
-            },
-            foreground = {
-                result.value = Consumable(databaseIds.count())
+        changeLockProvider.getInstanceLock(projectId).withLock { acquiredLock: Boolean ->
```

Comment:
I feel like this check would make more sense in `InstancesDataService#deleteInstance` (maybe in a modified method that takes a list of IDs rather just one). Then, the other places that delete instances can just call that and we'll only have to handle the check in one place.

Reply:
Done.